### DISCUSSION TITLE
Can the document TOC render HTML?

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#10158] [Q&A] [answered]
It would be great if I could do something like this:

![image](https://github.com/facebook/docusaurus/assets/47085752/aaab0e98-b079-417f-9ddd-7f040df19f43)

and have the TOC render it as well, so readers can immediately see what sections of a document are work in progress.

![image](https://github.com/facebook/docusaurus/assets/47085752/3e552330-57e4-4073-bb9d-fe9311a9d29b)

Is there a way to make this work? I think this would be good as default behavior but maybe that's just me 🤷‍♂️

**Comment by slorber** [comment, marked as answer]
The TOC headings are historically converted as strings

It is possible in the future that we'll support JSX in TOC better in the future, as explained in this PR comment: https://github.com/facebook/docusaurus/pull/9773#pullrequestreview-1872088268

---

However, I believe what you want to achieve is somehow possible already with some limitations, considering we use `dangerouslySetInnerHTML` from React you could use simple HTML tags in headings

An example is the RN website.

https://reactnative.dev/docs/accessibilityinfo#getrecommendedtimeoutmillis-android

```md
### `getRecommendedTimeoutMillis()` <div class="label android">Android</div>
```

![CleanShot 2024-05-23 at 10 47 21](https://github.com/facebook/docusaurus/assets/749374/2a015eb7-cd9a-4ef3-8947-db586071a9bf)

It's open source so you can try to copy their html/CSS and adjust this to your needs

**Reply by ZenaMel** [reply]
Thanks a lot @slorber
